ui/extension/statusPanel.tsx

```diff
@@ -0,0 +1,166 @@
+import React from 'react';
+import axios from 'axios';
+import { getPromotionStatus } from '../shared/src/utils/PSData';
+import { StatusIcon } from '../components-lib/src/components/StatusIcon';
+import type { StatusType } from '../components-lib/src/components/StatusIcon';
+import type { PromotionPhase, PromotionStrategy } from '../shared/src/types/promotion';
+
+// Types
+interface Application {
+  metadata: {
+    name: string;
+    namespace: string;
+  };
+  status?: {
+    resources?: Array<{
+      kind: string;
+      group: string;
+      name: string;
+      namespace: string;
+      status: string;
+    }>;
+  };
+}
+
+interface Resource {
+  kind: string;
+  name: string;
+  namespace: string;
+  liveState?: string;
+}
+
+const findPromotionStrategy = (application: Application) => {
```

Comment:
We should handle the case of multiple promotion stratigies in an application via just displaying some error or message.